build.sh

```diff
@@ -0,0 +1,161 @@
+  if [ ! -z "$(ls -A "$INSTALLATION_DIRECTORY")" ]; then
+     echo $INSTALLATION_DIRECTORY is not empty!
+     echo "Please enter an empty or a new directory!"
+     exit 1
+  fi
+}
+
+################################################################################
+function printInfo {
+################################################################################
+echo ---------------------------------------------------------------------------
+echo $*
+echo ---------------------------------------------------------------------------
+}
+
+################################################################################
+# END DECLARATION FUNCTIONS / START OF SCRIPT
+################################################################################
+
+################################################################################
+# Test for commands used in this script
+################################################################################
+testForCommands="dirname git date java javac"
+
+for command in $testForCommands
+do 
+  $command --help >> /dev/null
+  if [ $? -ne 0 ]; then
+    echo "Error: command '$command' is not installed!"
+    exit 1
+  fi
+done
+
+################################################################################
+# Determine directory of script. 
+################################################################################
+ACTUAL_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+################################################################################
+# Check parameters
+################################################################################
+checkParameters $*
+
+################################################################################
+# Determine repo name 
+################################################################################
+REPO_URL=`git remote get-url origin`
+REPO_NAME=${REPO_URL##*/}
+
+printInfo Build microservice of $REPO_NAME at $INSTALLATION_DIRECTORY
+
+
+################################################################################
+# Build service
+################################################################################
+
+echo Build service...
+./gradlew -Prelease clean build
+
+
+echo Copy configuration to $INSTALLATION_DIRECTORY...
+find . -name application-default.properties -exec cp '{}' $INSTALLATION_DIRECTORY/application.properties \;
+
+echo Copy jar file to $INSTALLATION_DIRECTORY...
+find . -name "$REPO_NAME*.jar" -exec cp '{}' $INSTALLATION_DIRECTORY \;
+
+echo Create config directory
+mkdir $INSTALLATION_DIRECTORY/config
+
+echo Create lib directory
+mkdir $INSTALLATION_DIRECTORY/lib
+
+###############################################################################
+# Create run script
+################################################################################
+printInfo Create run script ...
+
+# Determine name of jar file.
+for file in $INSTALLATION_DIRECTORY/*.jar; do
+  jarFile=${file##*/}
```

Comment:
Not exactly sure how this works (bash is magic) but the variable does not seem to work for me, as my script is generated like this:

```bash
jarFile=*.jar
ACTUAL_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
cd "$ACTUAL_DIR"
java -cp ".:$jarFile" -Dloader.path="file://$ACTUAL_DIR/$jarFile,./lib/,." -jar $jarFile
```

And after starting, the container output is: `Error: Unable to access jarfile *.jar`

Reply:
Ok due to the problems I found an easier solution:
cd $INSTALLATION_DIRECTORY
# Determine name of jar file.
jarFile=(`ls $REPO_NAME*.jar`)

should solve the problem.